columnar/src/backend/columnar/columnar_planner_hook.c

```diff
@@ -212,6 +212,10 @@ PlanTreeMutator(Plan *node, void *context)
 
 				customScan->custom_private = lappend(customScan->custom_private, vectorizedAggregateExecution);
 			}
+			else
+			{
+				elog(ERROR, "Custom Scan type is not ColumnarScan.");
```

Comment:
doesn't this just cause the operation to fail in another way?

Reply:
No, without this operation will not  fail and it returns unchanged custom scan node (leaving custom aggregator node)

With this PR, exception will be raised that will be caught in planner hook and no plan will be modified.